flytekit/core/type_engine.py

```diff
@@ -783,8 +783,8 @@ def t1() -> DC:
         from flytekit.types.file import FlyteFile
         from flytekit.types.structured import StructuredDataset
 
-        # Handle Optional
-        if UnionTransformer.is_optional_type(python_type):
+        # Handle Optional and Union Types
+        if UnionTransformer.is_optional_type(python_type) or _is_union_type(python_type):
```

Comment:
if we have the second one we don't need the first one right?

Reply:
Yep, the second one is a superset. Thanks for reminding me.